IsogeomGenerator/generate_isogeom.py

```diff
@@ -0,0 +1,507 @@
+            -db my_isogeom/
+
+    (2) Generate a database in the default location using levels between 1.0
+        2e+4 that are spaced with a ratio of 20:
+
+        generate_isogeom visit meshfile my_data -gl ratio -lx 1.0 2.e4 -N 20
+
+    (3) Generate a database in the default location using 15 levels between 1.0
+        2e+4 that are spaced logarithmically:
+
+        generate_isogeom visit meshfile my_data -gl log -lx 1.0 2.e4 -N 15
+
+    (4) Generate a database in a folder called 'my_isogeom/' and read the level
+        information from a file in the current directory called 'levelfile':
+
+        generate_isogeom visit meshfile my_data -lf levelfile -db my_isogeom/
+    """
+    visit_parser = subparsers.add_parser('visit',
+                                         description=visit_description,
+                                         usage=visit_usage,
+                                         epilog=visit_examples,
+                                         formatter_class=argparse.RawDescriptionHelpFormatter)
+    set_visit_only_options(visit_parser)
+    set_shared_options(visit_parser)
+    visit_parser.set_defaults(which='visit')
+
+    # set moab only mode options
+    moab_description = """
+Only generate the DAGMC-compliant geometry with MOAB starting from the VisIt
+mesh file database.
+
+Levels information must be provided with either the -lf or -lv option.
+"""
+    moab_usage = 'generate_isogeom moab [-lf/-lv] [OPTIONS]'
+    moab_examples = """
+Example Usage:
+    (1) Create an isosurface geometry called 'my_isogeom.h5m' with assigned
+        level values of 0.1 0.4 and 1.0, and tag the surfaces with data for
+        vizualization (assume default database location):
+
+        generate_isogeom moab -lv 0.1 0.4 1.0 -g my_isogeom.h5m --viz
+
+    (2) Generate a geometry from a database located in 'my_isogeom/', read the
+        level info from a file called 'levelinfo', mutliply all data by a
+        factor of 2e4, and save the file as 'my_isogeom.vtk' in a new folder
+        called 'output_folder/':
+
+        generate_isogeom moab -db my_isogeom/ -lf levelinfo -n 2e4
+            -g my_isogeom.vtk -sp output_folder/
+
+    (3) Generate a geometry from a database in the default location, read
+        levels from a file called 'levelfile' located in the database, tag the
+        geometry two metadata tags called E1 and E2 with values of 1.0 and
+        10.0, respectively, and tag the geometry with the level information for
+        vizualization:
+
+        generate_isogeom moab -lf tmp/levelfile -t E1 1.0 -t E2 10.0 -v
+    """
+    moab_parser = subparsers.add_parser('moab',
+                                        description=moab_description,
+                                        usage=moab_usage,
+                                        epilog=moab_examples,
+                                        formatter_class=argparse.RawDescriptionHelpFormatter)
```

Comment:
```suggestion
                                        formatter_class
                                            =argparse.RawDescriptionHelpFormatter)
```
would that be better ?

Reply:
No- pretty sure that is also a PEP8 violation.